azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -0,0 +1,219 @@
+from azurelinuxagent.common.utils import shellutil, fileutil
+from azurelinuxagent.common.utils.networkutil import AddFirewallRules
+from azurelinuxagent.common.utils.shellutil import CommandError
+
+
+class PersistFirewallRulesHandler(object):
+
+    __SERVICE_FILE_CONTENT = """
+# This drop-in unit file was created by the Azure VM Agent.
+# Do not edit.
+[Unit]
+Description=Setup network rules for WALinuxAgent 
+Before=network-pre.target
+Wants=network-pre.target
+DefaultDependencies=no
+ConditionPathExists={0}
+
+[Service]
+Type=oneshot
+Environment="DST_IP={1}" "UID={2}" "WAIT={3}"
+ExecStart={4} {0} --dst_ip=${{DST_IP}} --uid=${{UID}} ${{WAIT}}
+RemainAfterExit=false
+
+[Install]
+WantedBy=network.target
+"""
+
+    __OVERRIDE_CONTENT = """
+# This drop-in unit file was created by the Azure VM Agent.
+# Do not edit.
+[Service]
+# The first line clears the old data, and the 2nd line overwrites it.
+Environment=
+Environment="DST_IP={0}" "UID={1}" "WAIT={2}"
+"""
+
+    _AGENT_NETWORK_SETUP_BIN_FILE = "waagent_network_setup.py"
+    _AGENT_NETWORK_SETUP_NAME_FORMAT = "{0}-network-setup.service"
+    _DROP_IN_ENV_FILE_NAME = "10-environment.conf"
+
+    _FIREWALLD_RUNNING_CMD = ["firewall-cmd", "--state"]
+
+    def __init__(self, dst_ip, uid):
+        """
+        This class deals with ensuring that Firewall rules are persisted over system reboots.
+        It tries to employ using Firewalld.service if present first as it already has provisions for persistent rules.
+        If not, it then creates a new agent-network-setup.service file and copy it over to the osutil.get_systemd_unit_file_install_path() dynamically
+        On top of it, on every service restart it ensures that the WireIP is overwritten and the new IP is blocked as well.
+        """
+        osutil = get_osutil()
+        self._network_setup_service_name = self._AGENT_NETWORK_SETUP_NAME_FORMAT.format(osutil.get_service_name())
+        self._is_systemd = CGroupsApi.is_systemd()
+        self._systemd_file_path = osutil.get_systemd_unit_file_install_path()
+        self._agent_network_setup_bin_file = os.path.join(osutil.get_agent_bin_path(), self._AGENT_NETWORK_SETUP_BIN_FILE)
+        self._dst_ip = dst_ip
+        self._uid = uid
+        self._wait = osutil.get_firewall_will_wait()
+
+    @staticmethod
+    def _is_firewall_service_running():
+        # Check if firewall-cmd can connect to the daemon
+        # https://docs.fedoraproject.org/en-US/Fedora/19/html/Security_Guide/sec-Check_if_firewalld_is_running.html
+        # Eg:    firewall-cmd --state
+        #           >   running
+        firewalld_state = PersistFirewallRulesHandler._FIREWALLD_RUNNING_CMD
+        try:
+            return shellutil.run_command(firewalld_state).rstrip() == "running"
+        except Exception as error:
+            logger.info("{0} command returned error/not running: {1}".format(' '.join(firewalld_state), ustr(error)))
+        return False
```

Comment:
On success returns a string (stdout) and on failure a boolean. Let's make it consistent

Reply:
thanks, i missed the comparison